src/celengine/octree.h

```diff
@@ -300,9 +300,9 @@ inline void DynamicOctree<OBJ, PREC>::rebuildAndSort(StaticOctree<OBJ, PREC>*& _
     OBJ* _firstObject = _sortedObjects;
 
     if (_objects != nullptr)
-        for (typename ObjectList::const_iterator iter = _objects->begin(); iter != _objects->end(); ++iter)
+        for (typename ObjectList::iterator iter = _objects->begin(); iter != _objects->end(); ++iter)
         {
-            *_sortedObjects++ = **iter;
+            new (_sortedObjects++) OBJ(std::move(**iter));
```

Comment:
and move `_sortedObjects++` out of `new`, because of UB you might have here either `_sortedObjects` or `_sortedObjects + 1` if i correctly understand all this postfix `++` crap.